git-repository/examples/init-non-bare-repo.rs

```diff
@@ -0,0 +1,33 @@
+// cargo run -p git-repository --example new
+
+use anyhow::Context;
+use git_repository as git;
+
+fn main() -> anyhow::Result<()> {
+    let work_dir = std::env::args_os()
+        .nth(1)
+        .context("First argument needs to be the directory to initialize the repository in")?;
+    let repo = git::init(work_dir)?;
```

Comment:
I like this!  

FWIW, I was noodling with this approach:
```
    let work_dir = match (env::args().nth(1)) {
        Some(dir_name) => PathBuf::from(dir_name),
        None => {
            let tmp = tempfile::TempDir::new()?;
            tmp.path().join("repo-non-bare")
        }
    };
```
but your approach is more like a what a developer would write.

Reply:
It's fair to offer defaults though in case it shouldn't be an error. Tempdirs I'd not want to use anymore though, as `gitoxide` should be safe to use for examples as well (as in, "won't destroy your data").

Just being able to run an example while allowing some modification with additional arguments might be a way to go, and then ideally it informs about the possibility to set a value by passing arguments. Maybe that's something to try as well.

The work we are doing here can definitely help to bring all `git-repository` examples to a decent level that application developers would want to match at least.